src/services/identity/SitesCacheService.ts

```diff
@@ -14,81 +18,232 @@ import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
 // Changes are polled at fixed intervals, stored in cache, and served
 // to avoid long load time from live querying.
 
+const DEFAULT_PAGE_PARAMS = {
+  per_page: GH_MAX_REPO_COUNT,
+  sort: "full_name",
+  page: 1,
+}
+
+type GitHubResponseHeaders = HeadersDefaults & {
+  link?: string
+}
+
+type Link = {
+  relid: "first" | "last" | "prev" | "next"
+  url: string
+  pagenum: number
+}
+
+type LinkSet = {
+  first?: Link
+  last?: Link
+  prev?: Link
+  next?: Link
+}
```

Comment:
we could simplify this with a type:

```ts
type LinkSest = { 
  [linkRelation: LinkRelation]?: Link
}
```

Reply:
Does this declaration work? eslint/typescript are not letting me do it 🤔

Reply:
i got the syntax wrong oops

```ts
type LinkRelation = "first" | "last" | "prev" | "next"

type LinkSet = {
  [k in LinkRelation]: Link
}
```

Reply:
Ah right, thanks! 😄 

Pushed commit to do that: https://github.com/isomerpages/isomercms-backend/pull/1239/commits/027bc2215b19ebeba5647f497b3aa462fe4547c7